hw3/exercise-3/src/components/Houses.js

```diff
@@ -0,0 +1,311 @@
+						if (hashMap.has("Bolton")) {
+							hashMap.set("Bolton", hashMap.get("Bolton") + 1);
+						} else {
+							hashMap.set("Bolton", 1);
+						}
+					} else if (family[i].includes("Naharis")) {
+						if (hashMap.has("Naharis")) {
+							hashMap.set("Naharis", hashMap.get("Naharis") + 1);
+						} else {
+							hashMap.set("Naharis", 1);
+						}
+					} else if (
+						family[i].includes("Lorathi") ||
+						family[i].includes("Lorath")
+					) {
+						if (hashMap.has("Lorath")) {
+							hashMap.set("Lorath", hashMap.get("Lorath") + 1);
+						} else {
+							hashMap.set("Lorath", 1);
+						}
+					} else if (family[i].includes("Sand")) {
+						if (hashMap.has("Sand")) {
+							hashMap.set("Sand", hashMap.get("Sand") + 1);
+						} else {
+							hashMap.set("Sand", 1);
+						}
+					} else if (family[i].includes("Worm")) {
+						if (hashMap.has("Worm")) {
+							hashMap.set("Worm", hashMap.get("Worm") + 1);
+						} else {
+							hashMap.set("Worm", 1);
+						}
+					} else if (family[i].includes("Qyburn")) {
+						if (hashMap.has("Qyburn")) {
+							hashMap.set("Qyburn", hashMap.get("Qyburn") + 1);
+						} else {
+							hashMap.set("Qyburn", 1);
+						}
+					} else if (family[i].includes("Bronn")) {
+						if (hashMap.has("Bronn")) {
+							hashMap.set("Bronn", hashMap.get("Bronn") + 1);
+						} else {
+							hashMap.set("Bronn", 1);
+						}
+					} else if (family[i].includes("Sparrow")) {
+						if (hashMap.has("Sparrow")) {
+							hashMap.set("Sparrow", hashMap.get("Sparrow") + 1);
+						} else {
+							hashMap.set("Sparrow", 1);
+						}
+					} else if (family[i].includes("Mormont")) {
+						if (hashMap.has("Mormont")) {
+							hashMap.set("Mormont", hashMap.get("Mormont") + 1);
+						} else {
+							hashMap.set("Mormont", 1);
+						}
+					} else if (family[i].includes("Viper")) {
+						if (hashMap.has("Viper")) {
+							hashMap.set("Viper", hashMap.get("Viper") + 1);
+						} else {
+							hashMap.set("Viper", 1);
+						}
+					} else if (
+						family[i].includes("Unknown") ||
+						family[i].includes("None") ||
+						family[i].includes("Unkown") ||
+						family[i].includes("")
+					) {
+						if (hashMap.has("Unknown")) {
+							hashMap.set("Unknown", hashMap.get("Unknown") + 1);
+						} else {
+							hashMap.set("Unknown", 1);
+						}
+					}
+				}
+
+				const labels = [];
+				const values = [];
+
+				for (let [key, value] of hashMap) {
+					labels.push(key);
+					values.push(value);
+				}
+				setHouse({ labels, values });
+	}
+
+	useEffect(() => {
+		fetchData();
+	}, []);
+
+	const donut_chart = {
+		labels: house.labels,
+		datasets: [
+			{
+				label: "houses",
+				data: house.values,
+				backgroundColor: backgroundColors,
+				borderColor: borderColors,
+			},
+		],
+	};
+
+	const options = {};
+
+	return (
+		<div className="display">
+			<h1>Donut Chart:</h1>
+			<div className="donut">
+				<Doughnut data={donut_chart} options={options} />
+			</div>
```

Comment:
Due to its complexity. You could extract the doughnut chart to its own component to separate it from the houses page component.